extensions/amp-lightbox/1.0/element.js

```diff
@@ -0,0 +1,50 @@
+import {toggle} from '#core/dom/style';
+import {dispatchCustomEvent, toggleAttribute} from '#core/dom';
+import {unmountAll} from '#core/dom/resource-container-helper';
+
+export {CSS as shadowCss} from './component.jss';
+export {BentoLightbox as Component} from './component';
+
+export const usesShadowDom = true;
+export const props = {
+  'animation': {attr: 'animation', media: true},
+  'closeButtonAs': {selector: '[slot="close-button"]', single: true, as: true},
+  'children': {passthrough: true},
+};
+
+/**
+ * @param {Element} element
+ * @param {boolean} isOpen
+ * @return {boolean}
+ */
+export function setElementOpen(element, isOpen) {
+  toggleAttribute(element, 'open', isOpen);
+  toggle(element, isOpen);
+  dispatchCustomEvent(element, isOpen ? 'open' : 'close');
+  if (!isOpen) {
+    // Unmount all children when the lightbox is closed. They will automatically
+    // remount when the lightbox is opened again.
+    // TODO(wg-bento): Investigate if this is only needed on the AMP layer.
+    unmountAll(element, /* includeSelf */ false);
+  }
+  return isOpen;
+}
+
+/**
+ * @param {!Element} element
+ * @param {boolean} wasOpen
+ * @return {boolean}
+ */
+export function toggleOnMutation(element, wasOpen) {
```

Comment:
imo this should be handled in the component.js file itself by:

* adding a prop for `open`
* accepting a prop called `open` in the preact component
* adding an effect that triggers the api.open()/api.close() whenever the prop changes

```
function BentoLightbox({open: isOpen}) {
  const open = useCallback(...)
  const close = useCallback(...)
  useImperativeHandle(ref, () => ({
    open, close
  }), [...]) 
  
  useEffect(() => {
    isOpen ? open() : close();
  }, [isOpen, open, close])
  return ()
}

```